av1an-cli/src/lib.rs

```diff
@@ -442,164 +442,212 @@ fn confirm(prompt: &str) -> io::Result<bool> {
   }
 }
 
-pub fn parse_cli(args: CliOpts) -> anyhow::Result<EncodeArgs> {
-  let temp = if let Some(path) = args.temp.as_ref() {
-    path.to_str().unwrap().to_owned()
-  } else {
-    format!(".{}", hash_path(args.input.as_path()))
-  };
+/// Given Folder and File paths as inputs
+/// Converts them all to file paths
+/// Converting only depth 1 of Folder paths
+pub fn convert_input(args_paths: Vec<PathBuf>) -> Vec<PathBuf> {
+  let mut validated_inputs: Vec<PathBuf> = Vec::with_capacity(args_paths.len());
+
+  for input_arg in args_paths {
+    if input_arg.is_file() {
+      validated_inputs.push(input_arg.to_path_buf())
+    } else if input_arg.is_dir() {
+      let files: Vec<PathBuf> = std::fs::read_dir(input_arg)
+        .unwrap()
+        .into_iter()
+        .filter_map(Result::ok)
+        .filter_map(|d| {
+          if let Ok(file_type) = d.file_type() {
+            if file_type.is_file() {
+              Some(d.path())
+            } else {
+              None
+            }
+          } else {
+            None
+          }
+        })
+        .collect();
+      validated_inputs.extend(files)
```

Comment:
Since this is an iterator, I think you can extend without an intermediate vector/allocation.